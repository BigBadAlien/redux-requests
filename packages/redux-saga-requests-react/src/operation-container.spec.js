import renderer from 'react-test-renderer';
import React from 'react';

import OperationContainer from './operation-container';

describe('OperationContainer', () => {
  it('renders loading and error', () => {
    const component = renderer.create(
      <OperationContainer operation={{ pending: 1, error: 'error' }}>
        {({ loading, error }) => (
          <div>
            {loading && 'loading'}
            {error}
          </div>
        )}
      </OperationContainer>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders loading and error of operation with requestKey', () => {
    const component = renderer.create(
      <OperationContainer
        requestKey="x"
        operation={{ x: { pending: 1, error: 'error' } }}
      >
        {({ loading, error }) => (
          <div>
            {loading && 'loading'}
            {error}
          </div>
        )}
      </OperationContainer>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('doesnt crush if an operation with a requestKey doesnt exist', () => {
    const component = renderer.create(
      <OperationContainer
        requestKey="y"
        operation={{ x: { pending: 1, error: 'error' } }}
      >
        {({ loading, error }) => (
          <div>
            {loading && 'loading'}
            {error}
          </div>
        )}
      </OperationContainer>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders custom component as prop with extra prop', () => {
    const OperationComponent = ({ loading, error, extra }) => (
      <div>
        {loading && 'loading'} {error} {extra}
      </div>
    );

    const component = renderer.create(
      <OperationContainer
        operation={{ pending: 1, error: 'error' }}
        extra="extra"
        component={OperationComponent}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
