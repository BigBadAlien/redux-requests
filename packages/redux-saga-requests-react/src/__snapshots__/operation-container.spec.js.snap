// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationContainer doesnt crush if an operation with a requestKey doesnt exist 1`] = `<div />`;

exports[`OperationContainer renders custom component as prop with extra prop 1`] = `
<div>
  loading
   
  error
   
  extra
</div>
`;

exports[`OperationContainer renders loading and error 1`] = `
<div>
  loading
  error
</div>
`;

exports[`OperationContainer renders loading and error of operation with requestKey 1`] = `
<div>
  loading
  error
</div>
`;
